hlsl-headers/include/vk/khr/cooperative_matrix.impl

```diff
@@ -0,0 +1,216 @@
+// Copyright (c) 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "common.hlsli"
+
+namespace vk {
+namespace khr {
+namespace internal {
```

Comment:
What is the best way to hid everything in this section so that it is not visible outside of the implementation of the main class. Normally, we would use an anonymous namespace in the `.cpp` file. However, we do not have that option in HLSL because we do not link object files. I could make them private static member of a class, and make the main class a friend.

Reply:
I decided to name all of the internal fuctions as `__builtin_spv_*` functions. This should be good enough. I was able to remove the internal namespace.

For a few things that the user will need to access, I introduced a spirv.h file. This will include spec defined enums that the user may need to use.